src/SourceBuild/content/build.proj

```diff
@@ -43,6 +53,132 @@
+    <ExtractDotNetSdkArchiveDependsOn Condition="'$(ArchiveExtension)' == '.zip'">$(ExtractDotNetSdkArchiveDependsOn);ExtractDotNetSdkZip</ExtractDotNetSdkArchiveDependsOn>
+    <ExtractDotNetSdkArchiveDependsOn Condition="'$(ArchiveExtension)' == '.tar.gz'">$(ExtractDotNetSdkArchiveDependsOn);ExtractDotNetSdkTarball</ExtractDotNetSdkArchiveDependsOn>
+  </PropertyGroup>
+
+  <Target Name="ExtractDotNetSdkArchive"
+          DependsOnTargets="$(ExtractDotNetSdkArchiveDependsOn)" />
+
+  <Target Name="ExtractDotNetSdkTarball">
+    <MakeDir Directories="$(DotNetInstallDir)" />
+    <Exec Command="tar -xzf @(SdkArchiveItem) -C $(DotNetInstallDir)" />
+  </Target>
+
+  <Target Name="ExtractDotNetSdkZip">
+    <MakeDir Directories="$(DotNetInstallDir)" />
+    <Unzip SourceFiles="@(SdkArchiveItem)" DestinationFolder="$(DotNetInstallDir)" />
+  </Target>
+
+  <Target Name="ExtractScenarioTestsPackage">
+    <ItemGroup>
+      <_ScenarioTestPackage Include="$(ArtifactsShippingPackagesDir)scenario-tests/**/*.nupkg" />
+    </ItemGroup>
+    
+    <Unzip SourceFiles="@(_ScenarioTestPackage)" DestinationFolder="$(ScenarioTestsPackageExtractDir)" />
+
+    <ItemGroup>
+      <_ScenarioTestDll Include="$(ScenarioTestsPackageExtractDir)tools/**/$(ScenarioTestsProjectName).dll" />
+    </ItemGroup>
+  </Target>
+
+  <PropertyGroup>
+    <RunTestsTargets>$(RunTestsTargets);RunScenarioTests</RunTestsTargets>
+  </PropertyGroup>
+
+  <Target Name="Test" DependsOnTargets="ExtractDotNetSdkArchive;ExtractScenarioTestsPackage;$(RunTestsTargets)" />
+
+  <Target Name="RunScenarioTests"
+          Condition="'$(DotNetTestProjectsToRun)' == '' or $(DotNetTestProjectsToRun.Contains('$(ScenarioTestsProjectName)'))">
+    <Message Text="Running scenario tests: $(TargetArchitecture) $(Configuration) $(TargetOS) $(TargetRid)" Importance="high" />
+
+    <ItemGroup>
+      <_DotNetPath Include="$(DotNetInstallDir)sdk/**/dotnet.dll" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <_CurrentDateTime>$([System.DateTime]::Now.ToString("yyyy-MM-dd_HH_mm_ss"))</_CurrentDateTime>
+      <_TestXmlOutputPath>$(ScenarioTestsResultsDir)$(_CurrentDateTime).xml</_TestXmlOutputPath>
+      <_ScenarioTestsArgs>--dotnet-root $(DotNetInstallDir) --xml $(_TestXmlOutputPath) --target-rid $(TargetRid) --test-root $(ScenarioTestsRunsDir)$(_CurrentDateTime) --no-cleanup --no-traits Category=MultiTFM</_ScenarioTestsArgs>
+      <_MSBuildSdksDir>/%(_DotNetPath.Directory)Sdks</_MSBuildSdksDir>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_ScenarioTestsEnvVars Include="LOCAL_FEED_PATH=$(ArtifactsShippingPackagesDir)" />
+      <_ScenarioTestsEnvVars Include="MSBuildSDKsPath=$(_MSBuildSdksDir)" />
+    </ItemGroup>
+
+    <MakeDir Directories="$(ScenarioTestsResultsDir);$(ScenarioTestsRunsDir)" />
+
+    <Exec Command="$(DotnetTool) @(_ScenarioTestDll) $(_ScenarioTestsArgs)"
+          EnvironmentVariables="@(_ScenarioTestsEnvVars)" />
+  </Target>
```

Comment:
This belongs into the scenario tests project and not into build.proj. Otherwise adding a new test project wouldn't scale well.